src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts

```diff
@@ -193,110 +246,93 @@ export function useSwapInputsController({
         }
       }
 
-      isFetching.value = false;
       quote.value = data;
-      runOnJS(updateQuoteStore)(data);
 
       if (!data || (data as QuoteError)?.error) {
         return;
       }
 
       if (inputAmount) {
-        const price = internalSelectedInputAsset.value?.nativePrice || internalSelectedInputAsset.value?.price?.value || 0;
+        const price = inputPrice || inputNativePrice.value;
         inputValues.modify(prev => {
+          updateNativePriceForAsset({ price, type: 'inputAsset' });
           return {
             ...prev,
-            inputAmount,
+            inputAmount: Number(inputAmount),
             inputNativeValue: inputAmount * price,
           };
         });
       }
 
       if (outputAmount) {
-        const price = internalSelectedOutputAsset.value?.nativePrice || internalSelectedOutputAsset.value?.price?.value || 0;
+        const price = outputPrice || outputNativePrice.value;
         inputValues.modify(prev => {
+          updateNativePriceForAsset({ price, type: 'outputAsset' });
           return {
             ...prev,
-            outputAmount,
+            outputAmount: Number(outputAmount),
             outputNativeValue: outputAmount * price,
           };
         });
       }
+
+      isFetching.value = false;
+      if (isQuoteStale.value !== 0) isQuoteStale.value = 0;
+
+      runOnJS(updateQuoteStore)(data);
     },
     [
+      inputNativePrice,
       inputValues,
-      internalSelectedInputAsset.value?.balance.amount,
-      internalSelectedInputAsset.value?.nativePrice,
-      internalSelectedInputAsset.value?.price?.value,
-      internalSelectedOutputAsset.value?.nativePrice,
-      internalSelectedOutputAsset.value?.price?.value,
+      internalSelectedInputAsset,
```

Comment:
why not keep the specific data we need in this function which is now just the `internalSelectedInputAsset.value?.balance.amount` ?